### PR TITLE
infinite progress bar starts life in the stopped animation state

### PR DESCRIPTION
**Pull request by steveoc64**
.. then needs a Start() (or Show() ) to kick it off.

Addresses issue https://github.com/fyne-io/fyne/issues/202

By the time this PR and 200 is merged, Im pretty sure that https://github.com/fyne-io/fyne/issues/168 is sorted now, after running it through memdebug

### CODE REVIEW

**Comment by andydotxyz** [comment]
If we were to go in this direction then I think there is a larger change needed... That is to call Show() an all widgets that are really being shown.
To have this progress bar require additional setup (first time only) seems un-intuitive.

**Comment by steveoc64** [comment]
true. I think moving the initial start() inside the widget might resolve this without the need to extend the base.

**Comment by steveoc64** [comment]
Updated - call start() inside the widget constructor.

Tested, memdebugged, all good, no API behaviour changes now.

I took the liberty of adding 2 progress bar tabs on fyne_demo - 1 on the 1st tab, and 1 on the last tab to prove that the behaviour is correct in both cases, with no special code needed to start the animation.

Window closing all good too.

File: cmd/fyne_demo/widget.go [hunk 1]
```diff
@@ -63,9 +63,10 @@ func Widget(app fyne.App) {
 	w := app.NewWindow("Widgets")
 
 	w.SetContent(widget.NewTabContainer(
+		widget.NewTabItem("Progress Tab 1", makeProgressTab()),
```
Comment (andydotxyz) [inline cmd/fyne_demo/widget.go]: Can this be rolled back too? It doesn’t add anything to what should be a widget demo window...

If we want to keep something in place for test purposes then it probably belongs in a unit test.

**Comment by paulhovey** [comment]
Love the simple change - logic bug I didn't think of when implementing!

**Comment by andydotxyz** [comment]
I was about to merge this, and then thought... what about if someone uses the "&ProgressBarInfinite{}" format...

They will get the same widget but without the animation.

Not sure what I think now! Should we merge and document that curiosity or find another way to fix the animation restart issue?

**Comment by steveoc64** [comment]
If someone uses the &ProgressBarInfinite{} format ... then yes, they will have to call Start() on it using this method.

Will have a think about it ...  I dont think there is an easy way to get around that, unless we bump fyne itself to always propagate a Show() to any widget when it is first made visisble.

**Comment by paulhovey** [comment]
This came up when I was messing around as part of #217 , and start() is called in CreateRenderer(), which seems like a good default-on placement.  It can't be displayed without a renderer!

**Comment by andydotxyz** [comment]
This is superseded by the now-landed #217

**Pull request closed without merging**